Fix: NRage plugin gets stuck in initialization in certain cases

pcca-matrix opened the pull request:
The function wscanf was incorrectly used to parse VID_ and PID_ values from a string, causing failures in device detection. wscanf expects input from stdin, whereas in this case, the device ID string is stored in memory.

Using swscanf ensures correct parsing from memory instead of waiting for stdin

This fixes the issue where XInput devices not being detected properly at plugin init.

### Proposed changes
Replace wscanf with swscanf to correctly extract the vendor and product IDs from the device ID string.

### Does this make breaking changes?
No

### Does this version of Project64 compile and run without issue?
Yes

pcca-matrix committed:
Fix: NRage plugin gets stuck in initialization in certain cases

The function wscanf was incorrectly used to parse VID_ and PID_ values from a string, causing failures in device detection.
wscanf expects input from stdin, whereas in this case, the device ID string is stored in memory.

Using swscanf ensures correct parsing from memory instead of waiting for stdin

This fixes the issue where XInput devices not being detected properly at plugin init.

Squall-Leonhart commented:
good catch, this must have been fucked in the nrage for pj64 refactor because i remember using this as is right from the DX SDK sample.

edit: or not, i see its been wrong since i merged the code into the original project

what control were you seeing this with, i never have personally.

pcca-matrix commented:
issue occurs when the plugin attempts to initialize in streaming mode (e.g., Moonlight) or when the emulator is launched as a subprocess of a launcher using an Xbox360 or Xbox1 controller.
 
haven’t tested with other controllers but the problem should occur with any XInput device in these scenarios.

Squall-Leonhart commented:
I personally sign off on this change @project64

pcca-matrix committed:
N-Rage: Fix memory leak in GetN64ButtonArrayFromXAnalog and GetN64ButtonNameFromButtonCode

Replaced dynamic allocations (new[]) with static buffers to prevent memory leaks.
